force-app/main/default/classes/PropertyService.cls

```diff
@@ -57,6 +57,53 @@ public with sharing class PropertyService {
     }
+
+
+    /**
+     *  Payments__c custom object with relationship to Tenant and Property object.
+There is fields:
+- Date__c
+- Amount__c
+
+Create batch which will go through for all records in Property object.
+It will check if Property isRent and isPaid:
+- if yes → it will check if property has related payments record in current month
+- If no → it will send email reminder to tenant
+     * 
+     * @author GulnazAtabaeva
+     */
+
+    public void execute(System.SchedulableContext sc) {
+        CheckForBillsEachMonth batch = new CheckForBillsEachMonth();
+        Database.executeBatch(batch, 200);
+    }
+    
+    public Database.QueryLocator start(Database.BatchableContext bc) {
+        
+        return Database.getQueryLocator('SELECT Id, Is_Rented__c, Is_Paid__c FROM Property__c Where Is_Rented__c=true ANd Is_Paid__c=false');
+    }
+    
+    public void execute(Database.BatchableContext bc, List<Property__c> properties) {
+        Map<Id, Property__c> propertyMap = new Map<Id, Property__c>(properties);
+        List<Payments__c> paymentList = [Select Tenants__r.Email, Tenants__r.LastName From Payments__c Where Property__c IN: propertyMap.keySet() AND Date__c!=LAST_N_DAYS:30];
```

Comment:
You are querying payment records, where Date is older than last 30 days AND where Property is not paid.

Let's say that we have Property = 1, it's being rented by Tenant = Belek, it has Is Paid = false, which means it's not paid for current month. You are querying for related payments of Property = 1, where Date is older than last 30 days, which means your list will contains something like (let's assume current month July):

1. Payment(Date = June, Property = 1, Tenant = Belek)
2. Payment(Date = May, Property = 1, Tenant = Belek)
3. Payment(Date = April, Property = 1, Tenant = Belek)
4. Payment(Date = March, Property = 1, Tenant = Belek)
5. Payment(Date = Feb, Property = 1, Tenant = Bektur) 
6. Payment(Date = Jan, Property = 1, Tenant = Bektur)

These payments are related to Property = 1, which means it can return payments of previous Tenants like Bektur.

You are iterating through that list and sending email for each Tenant in a list. It means you will send email 6 time:

1 iteration you will send to Belek, that he didn't pay this month. (OK)
2 iteration you will send to Belek again, that he didn't pay this month. (NOT OK to send exact the same email 2x)
3 iteration you will send to Belek again, that he didn't pay this month. (NOT OK to send exact the same email 3x)
4 iteration you will send to Belek again, that he didn't pay this month. (NOT OK to send exact the same email 4x)
5 iteration you will send to Bektur, that he didn't pay this month. (NOT OK to send email to previous Tenant) 
6 iteration you will send to Bektur again, that he didn't pay this month. (NOT OK to send email to previous Tenant 2x)

Reply:
Great explanation! I will consider it. Thank you.